<launch>
  <node pkg="trobot" type="trobot_node" name="trobot_node">
    <param name="trobotVersion"        value="TRD-S" />
    <param name="activeConfig"         value="gemini" />
    <param name="serialPortName"       value="/dev/ttyACM0" />
    <param name="wheelRadius"          value="0.062" />
    <param name="axleTrack"            value="0.400" />
    <param name="rotEncoderResolution" value="800" />
    <param name="wheelGearRatio"       value="0.0667" />
    <param name="coeffRpsToCmd"        value="3.84192" />
    <param name="velAccelMax"          value="6" />
    <param name="velDecelMax"          value="6" />
    <param name="autoRecovery"         value="AUTO" />
    <param name="FrameRate"            value="16" />
    <param name="publish_tf"           value="true" />
    <param name="odom_frame"           value="odom" />
    <param name="base_frame"           value="base_footprint" />
  </node>


  <node pkg="joy" type="joy_node" name="joy_node">
    <param name="dev" value="/dev/input/js0" />
    <param name="deadzone" value="0.3" />
    <param name="autorepeat_rate" value="16" />
  </node>

  <arg name="config_filepath" default="$(find trobot)/launch/joy_param.config.yaml" />

  <node pkg="teleop_twist_joy" name="teleop_twist_joy" type="teleop_node">
    <rosparam command="load" file="$(arg config_filepath)" />
  </node>
</launch>
